setup/src/Magento/Setup/Console/Command/ModuleStatusCommand.php

```diff
@@ -47,23 +56,101 @@ protected function configure()
      */
     protected function execute(InputInterface $input, OutputInterface $output)
     {
-        $moduleList = $this->objectManagerProvider->get()->create(\Magento\Framework\Module\ModuleList::class);
+        $moduleName = (string)$input->getArgument('module');
+        if ($moduleName) {
+            return $this->showSpecificModule($moduleName, $output);
+        }
+
+        $onlyEnabled = $input->getOption('enabled');
+        if ($onlyEnabled) {
+            return $this->showEnabledModules($output);
+        }
+
+        $onlyDisabled = $input->getOption('disabled');
+        if ($onlyDisabled) {
+            return $this->showDisabledModules($output);
+        }
+
         $output->writeln('<info>List of enabled modules:</info>');
-        $enabledModules = $moduleList->getNames();
-        if (count($enabledModules) === 0) {
+        $this->showEnabledModules($output);
+        $output->writeln('');
+
+        $output->writeln("<info>List of disabled modules:</info>");
+        $this->showDisabledModules($output);
+        $output->writeln('');
+    }
+
+    /**
+     * @param string $moduleName
+     * @param OutputInterface $output
+     */
+    private function showSpecificModule(string $moduleName, OutputInterface $output)
+    {
+        $allModules = $this->getAllModules();
+        if (!in_array($moduleName, $allModules->getNames())) {
+            $output->writeln('<error>Module does not exist</error>');
+            return;
```

Comment:
Maybe we can add `return \Magento\Framework\Console\Cli::RETURN_FAILURE;` here to indicate something went wrong similar to here 
https://github.com/jissereitsma/magento2/blob/047097beeed4d70f437270db34681f9022ffee67/setup/src/Magento/Setup/Console/Command/AbstractMaintenanceCommand.php#L86

Reply:
Good point. I was in a rush to finalize things, but did not like the return of no error nice anyway. This is fixed now in https://github.com/magento/magento2/pull/15543/commits/ca4282e5c95dd1436231aaa3f7c94995231d27a8